pygen/pygen_src/riscv_asm_program_gen.py

```diff
@@ -629,7 +629,7 @@ def generate_directed_instr_stream(self, hart = 0, label = "", original_instr_cn
                 if(object_h is None):
                     logging.critical("Cannot create instr stream %0s", name)
                     sys.exit(1)
-                new_instr_stream = copy.copy(object_h)
+                new_instr_stream = copy.deepcopy(object_h)
```

Comment:
is this going to incur significant performance penalties for tests that use a lot of directed streams?

Reply:
Yes, there will be some performance issues as memory usage will be more.
To fix the label issue we have used the deep copy here.

Reply:
that makes sense to me, we can always use a workaround if the performance becomes unacceptable.